### ISSUE TITLE
format_help_text does not format text in a group

### ISSUE THREAD

**Issue by jonathan-s** [pallets/click#2688]
You can override the function `format_help_text` to modify the help output of the command. However this does _not_ affect the formatting of the help when it is displayed in a `group` setting. 

`MultiCommand`, format the text in `format_commands` and that only uses `get_short_help_str`, it seems like `get_short_help_str` should be using `format_help_text` to get shortened help text, nothing else. 

```
class HelpCommand(click.Command):
    def format_help_text(self, ctx, formatter):
        formatter.write("New string")
        

@click.command(cls=HelpCommand)  
def test():
    """Old string"""
    pass
    
@click.group()
def cli():
    """Cli command"""
    pass
```

When calling cli it will display "Old string", when calling test individually it will display "New string"

Environment:
click 8.1.7

**Comment by davidism** [comment]
I can't reproduce this issue with the information provided. It's no possible to call `test` through the `cli` group because it's not registered as part of the group.

**Comment by davidism** [comment]
I'm guessing you're saying that in the list of subcommands, it shows the docstring instead of the output of your `format_help_text`. But that's because they're different cases, `format_help_text` is used when formatting the command's own help text. You either need to set `short_help` on the command or override `get_short_help_str`.

**Comment by jonathan-s** [comment]
@davidism They are not totally different cases though as setting the docstrings populates both of them. I found it kind of confusing that you'd have to override both methods. 

Sure if explicitly set `short_help` I would agree that it should use that explicit value otherwise it should use a short version of `format_help_text`.

**Comment by davidism** [comment]
Yeah, setting the docstring sets the default value for `help` and `short_help`. You're overriding the formatting of `help`, but you're doing nothing to `short_help`. They're different things.
